apps/meteor/app/api/server/v1/channels.ts

```diff
@@ -954,7 +951,12 @@ API.v1.addRoute(
 
 API.v1.addRoute(
 	'channels.list',
-	{ authRequired: true },
+	{
+		authRequired: true,
+		permissionsRequired: {
+			GET: { permissions: ['view-c-room', 'view-joined-room'], operation: 'hasAny' },
```

Comment:
Didn't see any use of `view-joined-room` 🤔

Reply:
It's inside `if (!hasPermissionToSeeAllPublicChannels) {`
The user must have at least one of the permissions